cmd/kubeadm/app/apis/kubeadm/v1beta3/conversion.go

```diff
@@ -36,3 +36,19 @@ func Convert_v1beta3_InitConfiguration_To_kubeadm_InitConfiguration(in *InitConf
 	SetDefaults_ClusterConfiguration(clusterCfg)
 	return Convert_v1beta3_ClusterConfiguration_To_kubeadm_ClusterConfiguration(clusterCfg, &out.ClusterConfiguration, s)
 }
+
+func Convert_kubeadm_DNS_To_v1beta3_DNS(in *kubeadm.DNS, out *DNS, s conversion.Scope) error {
+	return autoConvert_kubeadm_DNS_To_v1beta3_DNS(in, out, s)
+}
+
+// Convert_v1beta3_ClusterConfiguration_To_kubeadm_ClusterConfiguration is an autogenerated conversion function.
+func Convert_v1beta3_ClusterConfiguration_To_kubeadm_ClusterConfiguration(in *ClusterConfiguration, out *kubeadm.ClusterConfiguration, s conversion.Scope) error {
+	err := autoConvert_v1beta3_ClusterConfiguration_To_kubeadm_ClusterConfiguration(in, out, s)
+	if err != nil {
+		return err
+	}
+	// make roundrip with v1beta2 happy by setting the v1beta2 default here
+	// TODO: remove with v1beta2 https://github.com/kubernetes/kubeadm/issues/2459
+	out.DNS.Type = "CoreDNS"
```

Comment:
i guess this can just be `kubeadm.CoreDNS` instead of using this literal string.